scripts/scraper-ng/test/ingredient-data.formal_syntax.test.js

```diff
@@ -0,0 +1,73 @@
+const {
+  process,
+  expectPositionElement,
+  expectNullPosition,
+} = require("./framework/utils");
+
+describe("data.formal_syntax", () => {
+  const ingredientName = "data.formal_syntax";
+  const recipe = { body: [ingredientName] };
+
+  test("valid", () => {
+    const valid = `<h2 id="Formal_syntax">Formal syntax</h2>
+                   <pre class="syntaxbox notranslate">{{ csssyntax("display") }}</pre>`;
+    const file = process(valid, recipe);
+
+    expect(file).not.hasMessageWithId(ingredientName);
+    expectPositionElement(file.data.ingredients[0], ingredientName, "h2");
+  });
+
+  test("missing h2", () => {
+    const file = process("", recipe);
+
+    expect(file.messages.length).toBeGreaterThan(0);
+    expect(file).hasMessageWithId(/data.formal_syntax\/expected-heading/);
+
+    expectNullPosition(file.data.ingredients[0], ingredientName);
+  });
+
+  test("missing pre", () => {
+    const missingPre = `<h2 id="Formal_syntax">Formal syntax</h2>
+                        {{ csssyntax("display") }}`;
+    const file = process(missingPre, recipe);
+
+    expect(file.messages.length).toBeGreaterThan(0);
+    expect(file).hasMessageWithId(/data.formal_syntax\/expected-pre.syntaxbox/);
+
+    expectNullPosition(file.data.ingredients[0], ingredientName);
+  });
+
+  test("missing syntaxbox class", () => {
+    const missingClass = `<h2 id="Formal_syntax">Formal syntax</h2>
+                          <pre>{{ csssyntax("display") }}</pre>`;
+    const file = process(missingClass, recipe);
+
+    expect(file.messages.length).toBeGreaterThan(1);
```

Comment:
Here we check `toBeGreaterThan(1)`. So because there's an error and a CSSSyntax macro call, then there are two messages. But in the previous test ("missing pre"), the same applies but we only check `toBeGreaterThan(0)`. I don't understand why.

I also don't understand, if we're going to check the number of messages, why not check the exact number rather than the weaker "greater than". But in general, I don't think it's worth checking the number of errors here at all.